### PR TITLE
Fixed compile errors when building on zLinux

### PR DESCRIPTION
**Pull request by exinfinitum**
Added/modified several #ifdefs that check for whether the system OS is zOS.

Also fixed several places where zLinux compiler complained that parentheses were needed around boolean statements, in runtime.cc.

### CODE REVIEW

File: src/log-utils.cc [hunk 1]
```diff
@@ -186,7 +186,12 @@ void Log::MessageBuilder::AppendDetailed(String* str, bool show_impl_info) {
   }
   for (int i = 0; i < len; i++) {
     uc32 c = str->Get(i);
-    if (ebcdic2ascii(c) > 0xff) {
+#ifdef V8_OS_ZOS
+    bool CharGreaterThanFF = ebcdic2ascii(c) > 0xff;
+#else
+    bool CharGreaterThanFF = false;
+#endif
+    if (CharGreaterThanFF) {
```
Comment (mcornac) [inline src/log-utils.cc]: I think this should be
#ifdef V8_OS_ZOS
if (ebcdic2ascii(c) > 0xff) {
#else
if (c > 0xff) {
#endif

And it looks like its needed 2 other places from the commit that added ebcdic support here:
https://github.com/ibmruntimes/v8z/commit/93931240f189f38634b5d3ff7de18f42d4adbe3b

File: src/utils.h [hunk 2]
```diff
@@ -196,9 +196,10 @@ inline const uint8_t& ebcdic2ascii(const char letter) {
   };
   return e2a[letter];
 #else
-  return letter;
+  const uint8_t* letter_to_return = reinterpret_cast<const uint8_t*>(&letter);
+  return *letter_to_return;
 #endif
-};
+}
```
Comment (mcornac) [inline src/utils.h:198]: Can you try to #ifdef the whole ebcdic2ascii function. I don't think it should exist on zLinux and that will make it easier to find bugs.

Comment (joransiu) [inline src/utils.h:198]: Should ebcdic2ascii take and return the same type?

Comment (mcornac) [inline src/utils.h:198]: cc @mmallick-ca any input on the return type of ebcdic2ascii?

Comment (mmallick-ca) [inline src/utils.h:198]: I believe I have all ready committed the change which would make ebcdic2ascii function a nop for any platform other than zos

Comment (joransiu) [inline src/utils.h:198]: @mmallick-ca I meant to ask why ebcdic2ascii takes a char, and returns an uint_8.

File: src/log-utils.cc [hunk 3]
```diff
@@ -187,14 +187,16 @@ void Log::MessageBuilder::AppendDetailed(String* str, bool show_impl_info) {
   for (int i = 0; i < len; i++) {
     uc32 c = str->Get(i);
 #ifdef V8_OS_ZOS
-    bool CharGreaterThanFF = ebcdic2ascii(c) > 0xff;
-#else
-    bool CharGreaterThanFF = false;
-#endif
-    if (CharGreaterThanFF) {
+    if (ebcdic2ascii(c) > 0xff) {
```
Comment (joransiu) [inline src/log-utils.cc:190]: @mmallick-ca What should be the proper behaviour here?   ebcdic2ascii will return uint8_t, so it'll never be greater than 0xff.

Comment (mmallick-ca) [inline src/log-utils.cc:190]: @joransiu Yeah you are right, I think we need to do a range check on c first prior to the ascii conversion. if  c is in a valid ebcdic range than we should convert it

File: tools/testrunner/local/commands.py [hunk 4]
```diff
@@ -44,7 +45,16 @@ def KillProcessWithID(pid):
...
+  if (platform.system() == 'OS/390'):
+    os.system("touch /tmp/cleanup_semaphores.sh")
+    os.system("chmod a+rx /tmp/cleanup_semaphores.sh")
+    os.system("echo '#!/bin/bash' >> /tmp/cleanup_semaphores.sh")
+    os.system("echo 'ME=`whoami`' >> /tmp/cleanup_semaphores.sh")
+    os.system("echo 'for u in `ipcs -s | grep $ME | tr -s'\" ' '\"'| cut -d'\"'"
+              " '\"' -f2` ; do ipcrm -s  $u; done'>>/tmp/cleanup_semaphores.sh")
+    os.system("cat /tmp/cleanup_semaphores.sh")
+    os.system("/tmp/cleanup_semaphores.sh")
+    os.system("rm /tmp/cleanup_semaphores.sh")
```
Comment (mcornac) [inline tools/testrunner/local/commands.py]: Does it work to just execute the shell command directly instead? I.e.:
os.system("for u in $(ipcs -s | grep $(whoami) | tr -s'\" ' '\"'| cut -d'\"'" " '\"' -f2) ; do ipcrm -s  $u; done")

Comment (mmallick-ca) [inline tools/testrunner/local/commands.py]: @exinfinitum : looks good to me, I would suggest that you do not create and delete the cleanup file on every invocation of CleanupSemaphores. Maybe generate cleanup shell script somewhere else, invoke it in CleanupSemaphores and remove it once the test run is complete.

Comment (mcornac) [inline tools/testrunner/local/commands.py]: os.system("for u in $(ipcs -s | grep $(whoami) | tr -s ' ' | cut -d ' ' -f2) ; do ipcrm -s $u; done");
Seems to work for me.

**Comment by exinfinitum** [comment]
Merged as cc4709326f224a1ec8799a77cd7a99e162d73f22.
Done manually due to merge conflict in src/utils.h.

**Pull request closed without merging**